### PR TITLE
Keep error reporter open if it fails to send report

### LINKED ISSUE #33269: Error reporter should stay open if it fails to send
**Original reporter:** Found in https://github.com/mantidproject/mantid/issues/33200

**Describe the bug**
If the error report fails to send, a pop-up saying to contact our help email appears. Clicking OK closes the pop-up AND the error report. The error reporter should not close so that the user can extract any information from the stacktrace or the textbox they've filled out. The pop-up should maybe also point to the forum instead of the email and tell the user to close the error reporter by clicking "Don't share any information".

**To Reproduce**
- Locally undo the PR fix here: https://github.com/mantidproject/mantid/issues/33219
- Then raise an error reporter by running segfault (dry-run = false) or using the soft crash option here: https://developer.mantidproject.org/Testing/ErrorReporter-ProjectRecovery/ErrorReporterTesting.html
- Then input `"` in the name, email and textbox fields and click Yes, share information. You will get a bad request 400 and the pop-up described above! Click OK and the pop-up and error reporter close!

**Expected behavior**
Pop-up closes, errorreporter stay open.
The pop-up should maybe also point to the forum instead of the email and tell the user to close the error reporter by clicking "Don't share any information".

**Screenshots**
Screenshot 2022-01-06 at 11 46 47 (2)

**Platform/Version (please complete the following information):**
 - OS: all OS
 - OS Version:
 - Mantid Version nightly 4th Jan 2022

### PR DESCRIPTION
**Pull request by jhaigh0**
**Description of work.**

Before, if an error report failed to send, the error reporter would still close. This is annoying as it looses all the stack trace and comment data. In this pr the error reporter will only close if the report sends successfully or the user chooses to not send a report.

**To test:**

- [x] Add the line `errorreports.rooturl = https://error.mantidproject.org` to your `Mantid.user.properties` file (this url is incorect so will cause the error reporter to fail to send).

My preferred method for bringing up the error reporter is to first remove the `--single-process` argument from my PyCharm configuration, then launch Mantid and run the SegFault algorithm. Alternatively, you can follow test 3 [here](https://developer.mantidproject.org/Testing/ErrorReporter-ProjectRecovery/ErrorReporterTesting.html).

- [x] Click *Yes, share information*
  - [x] An error warning should appear an upon closing it the error reporter should stay open
-  [x] Click *Share non-identifiable information*
  - [x] The same should happen
- [x] Click *Don't share any information*
  - [x] The error reporter should close

Fixes #33269

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by sf1919** [comment]
I followed the instructions but I never got the error reporter. If I turn off Dryrun on the segfault algorithm then the whole thing crashes and closes. If I leave Dryrun ticked I get the crash error and Mantid doesn't close but I don't get an errror reporter either.

**Comment by jhaigh0** [comment]
> I followed the instructions but I never got the error reporter. If I turn off Dryrun on the segfault algorithm then the whole thing crashes and closes. If I leave Dryrun ticked I get the crash error and Mantid doesn't close but I don't get an errror reporter either.

You could try number three [here](https://developer.mantidproject.org/Testing/ErrorReporter-ProjectRecovery/ErrorReporterTesting.html)

File: docs/source/release/v6.6.0/Workbench/Bugfixes/33269.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+- If the error reporter fails to send, it will now not automatically close
```
Comment (ConorMFinn) [inline docs/source/release/v6.6.0/Workbench/Bugfixes/33269.rst:1]: I would re-word this to say what the reporter does now, maybe this?:
```suggestion
- The error reporter will now stay open and provide an error message if the report failed to send. 
```

**Review by cailafinn — CHANGES_REQUESTED** [review]
Works well for me after enabling usage reporting and restarting workbench. Just a quick comment on the release note.

File: docs/source/release/v6.6.0/Workbench/Bugfixes/33269.rst (hunk 1 above)
Comment (jhaigh0) [reply docs/source/release/v6.6.0/Workbench/Bugfixes/33269.rst:1]: Much better wording thank you

**Review by cailafinn — APPROVED** [review]
LGTM! :shipit:

**Pull request merged**